src/wp-login.php

```diff
@@ -206,7 +206,7 @@ function login_header( $title = 'Log In', $message = '', $wp_error = null ) {
 
 	?>
 	<div id="login">
-		<h1><a href="<?php echo esc_url( $login_header_url ); ?>"><?php echo $login_header_text; ?></a></h1>
+		<h1><a href="<?php echo esc_url( $login_header_url ); ?>"><?php echo esc_html( $login_header_text ); ?></a></h1>
```

Comment:
I'm not sure `esc_html()` is the right function to use here. This is inside an `a` element which has a [transparent content model](https://html.spec.whatwg.org/multipage/dom.html#transparent-content-models):

> [Transparent](https://html.spec.whatwg.org/multipage/dom.html#transparent-content-models), but there must be no interactive descendant, `a` element descendant, or descendant with the `tabindex` attribute specified.
> [ref](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element)

As it's transparent, valid content matches that of its parent, a `h1` element (with exceptions as noted above).

The `h1` element has a [phrasing content model](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content).

Escaping all HTML could therefore break BC if folks have used something like:

```php
get_bloginfo( 'title' ) . '<sup>&trade;</sup>'

```

That said, it doesn't mean we need to support everything here, but it doesn't seem like we should be blocking something like the simple example above.

Reply:
Thank you for reviewing the changes.

Yes, I had the same thought as you, and I also wanted to use `esc_js()` or a selective KSES function. However, I was uncertain about which one would be more appropriate.

Reply:
Thank you for reviewing the changes.

Yes, I had the same thought as you, and I also wanted to use esc_js() or a selective KSES function. However, I was uncertain about which one would be more appropriate.